### PR TITLE
#3 : Navbar

### LINKED ISSUE #3: Navbar
> Note: For the mobile version you can also have a look at (and even just copy aspects of) the navbar implemented for the [Seneca Innovators Website](https://github.com/Seneca-Software-Developers-Club/seneca-innovators-website) as long as it suits the aesthetics of the website.

- You need to work with `components/Navbar.tsx` and might benefit from making another component/function for a navbar link/button within the same file, for better readability.
- If you feel that a UI library like DaisyUI (or any others) would make this cleaner and easier, feel free to bring it up in a comment here and use it if others agree.
 
- The navbar needs to have all of the links except for `Resources`
Image

- The mobile version has the hamburger menu at the bottom-left, change it to be on the top-right. The exact manner in which the navbar pops open is up to you.
Image

### PR DESCRIPTION
**Pull request by krutinpolra**
This pull request introduces a fully responsive Navbar component with a dynamic dropdown menu for mobile screens as well. The navbar has been designed to improve navigation across all screen sizes while maintaining a consistent and sleek appearance.

Desktop:
image

Mobile:
image
image

Issue: #3

### CODE REVIEW

**Review by CrisH2307 — DISMISSED** [review]
Looks good, one more thing can be improved in the future (optional):
The alignment of the nav tab could be prettier if it can be justified in the center
Screenshot 2024-12-24 at 1 56 18 PM

Overall, this is good to merge.

**Comment by krutinpolra** [comment]
> This pull request introduces a fully responsive Navbar component with a dynamic dropdown menu for mobile screens as well. The navbar has been designed to improve navigation across all screen sizes while maintaining a consistent and sleek appearance.
> 
> Desktop: image
> 
> Mobile: image image
> 
> Issue: #3

Here is the improve version of navigation items within the dropdown menu for mobile screens. The dropdown menu now centers its items both vertically and horizontally, ensuring a clean and visually appealing layout.

image

File: components/Navbar.tsx [hunk 1]
```diff
@@ -1,3 +1,89 @@
...
+          aria-label="Toggle Menu"
+          onClick={() => setMenuOpen(!menuOpen)}
+        >
+          <Bars3Icon className="h-8 w-8" />
+        </button>
+
+        {/* Desktop navigation menu */}
+        <ul className="hidden md:flex space-x-6 lg:space-x-8 items-center">
+          {navItems.map((item) => (
+            <li key={item.label}>
```
Comment (KaosElegent) [inline components/Navbar.tsx:45]: I like the use of `item.label` as the key instead of an `index`

**Review by KaosElegent — APPROVED** [review]
Looks good!

**Pull request merged**